src/image-extension-service.ts

```diff
@@ -44,17 +46,38 @@ export class ImageExtensionService {
       const assets = this._googleAdsApi
         .getAssetsForAdGroup(adGroup.adGroup.id)
         .filter(e => uploadedToGcsImages?.includes(e.name));
-      const notLinkedAssets = assets.filter(e => !e.adGroupAssetResourceName);
-
-      // Adding images from GCS to Ad Groups
-      Logger.log(
-        `Creating ${notLinkedAssets.length} ad group assets for ad group ${adGroup.adGroup.id}...`
+      const existingAssets = this._googleAdsApi.getAllAdGroupAssetsForAdGroup(
+        adGroup.adGroup.id
       );
-      this._googleAdsApi.createAdGroupAssets(
-        adGroup.adGroup.resourceName,
-        notLinkedAssets
+      const notLinkedAssets = assets
```

Comment:
Please consider moving this to the pipeline on line 46 (just adding these filter and sort there)

Reply:
Done